### DISCUSSION TITLE
NFS write error occurring randomly only on Clients

### DISCUSSION THREAD

**Discussion by knightflyer** [ltsp/ltsp#608] [Q&A]
Using WD SSD drives
Mint Mate 20.2
Latest ltsp binaries
Network of 6 desktops running ThinClient
When system becomes sightly or moderately active some  " lower powered clients" announce write error on "invoice" file.  The file is not corrupted and after several tries data write may become successful.  This does not occur on the server and does not occur on traditional nfs mounted clients (NON-LTSP clients)

following is my "/etc/exports" from the server
/u    192.168.1.0/24(rw,sync,all_squash,anonuid=1000,anongid=1010,no_subtree_check)
following is my fstab declaration in /etc/ltsp.conf
FSTAB_x="192.168.1.200:/u /u nfs"

I don't want to go back to Fat Clients, but I have to make decision before deployment next week.
Any ideas? the client's fstab entry looks kind of light to me.

Thanks Everyone

**Comment by alkisg** [comment]
In the LTSP terminology, "thin clients" refer to "netbooted PCs that use remote desktop to the server",
"fat clients" to "netbooted PCs that use network disks but run the apps locally",
and "standalone clients" to "PCs that boot from local hard disks".

So I guess you're saying that you're running LTSP fat clients and you don't want to go back to standalone clients.

Boot an LTSP fat client, login as a user, type `cat /proc/mounts`, and paste the output here.
Then boot a standalone client, login as a user, type `cat /proc/mounts`, and paste the output here.

It's possible that your issue is with the SSHFS mounting of /home, and not your /u netmounted NFS directory.

**Comment by knightflyer** [comment]
*Output from "Fat Client" (Client boots from network getting instructions
from ltsp.conf)*
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
udev /dev devtmpfs
rw,nosuid,noexec,relatime,size=3970140k,nr_inodes=992535,mode
=755 0 0
devpts /dev/pts devpts
rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,nosuid,nodev,noexec,relatime,size=803612k,mode=755 0 0
tmpfs /run/initramfs/ltsp tmpfs rw,relatime,mode=755 0 0
/dev/loop0 /run/initramfs/ltsp/0/looproot squashfs ro,relatime 0 0
/run/initramfs/ltsp / overlay
rw,relatime,lowerdir=/run/initramfs/ltsp/0/looproo
t,upperdir=/run/initramfs/ltsp/0/up,workdir=/run/initramfs/ltsp/0/work 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime
0 0
tmpfs /dev/shm tmpfs rw,nosuid,nodev 0 0
tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime,size=5120k 0 0
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup2 /sys/fs/cgroup/unified cgroup2
rw,nosuid,nodev,noexec,relatime,nsdelegat
e 0 0
cgroup /sys/fs/cgroup/systemd cgroup
rw,nosuid,nodev,noexec,relatime,xattr,name=
systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
none /sys/fs/bpf bpf rw,nosuid,nodev,noexec,relatime,mode=700 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0
0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory
0 0
cgroup /sys/fs/cgroup/perf_event cgroup
rw,nosuid,nodev,noexec,relatime,perf_eve
nt 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup
rw,nosuid,nodev,noexec,relatime,cpu,cpu
acct 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
cgroup /sys/fs/cgroup/rdma cgroup rw,nosuid,nodev,noexec,relatime,rdma 0 0
cgroup /sys/fs/cgroup/devices cgroup
rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup
rw,nosuid,nodev,noexec,relatime,ne
t_cls,net_prio 0 0
cgroup /sys/fs/cgroup/freezer cgroup
rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset
0 0
cgroup /sys/fs/cgroup/hugetlb cgroup
rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs
rw,relatime,fd=28,pgrp=1,timeout=0,min
proto=5,maxproto=5,direct,pipe_ino=23579 0 0
mqueue /dev/mqueue mqueue rw,nosuid,nodev,noexec,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,relatime,pagesize=2M 0 0
sunrpc /run/rpc_pipefs rpc_pipefs rw,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,nosuid,nodev,noexec,relatime 0 0
tracefs /sys/kernel/tracing tracefs rw,nosuid,nodev,noexec,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,nosuid,nodev,noexec,relatime 0
0
configfs /sys/kernel/config configfs rw,nosuid,nodev,noexec,relatime 0 0
192.168.1.200:/u /u nfs4
rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen
=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.1.13,local_lo
ck=none,addr=192.168.1.200 0 0
tmpfs /run/user/1009 tmpfs
rw,nosuid,nodev,relatime,size=803608k,mode=700,uid=10
09,gid=1009 0 0
***@***.***: /home/randy fuse.sshfs
rw,nosuid,nodev,relatime,user_id=0,group_id
=0,default_permissions,allow_other 0 0
gvfsd-fuse /run/user/1009/gvfs fuse.gvfsd-fuse
rw,nosuid,nodev,relatime,user_id=
1009,group_id=1009 0 0
/dev/fuse /run/user/1009/doc fuse
rw,nosuid,nodev,relatime,user_id=1009,group_id
=1009 0 0





*Output from Stand Alone Client (Client boots off his own hard drive and
connects via NFS solely)*
tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime,size=5120k 0 0
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup2 /sys/fs/cgroup/unified cgroup2
rw,nosuid,nodev,noexec,relatime,nsdelegate 0 0
cgroup /sys/fs/cgroup/systemd cgroup
rw,nosuid,nodev,noexec,relatime,xattr,name=systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
none /sys/fs/bpf bpf rw,nosuid,nodev,noexec,relatime,mode=700 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup
rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0
0
cgroup /sys/fs/cgroup/devices cgroup
rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset
0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory
0 0
cgroup /sys/fs/cgroup/rdma cgroup rw,nosuid,nodev,noexec,relatime,rdma 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup
rw,nosuid,nodev,noexec,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/perf_event cgroup
rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/freezer cgroup
rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup
rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs
rw,relatime,fd=28,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=22611
0 0
mqueue /dev/mqueue mqueue rw,nosuid,nodev,noexec,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,relatime,pagesize=2M 0 0
debugfs /sys/kernel/debug debugfs rw,nosuid,nodev,noexec,relatime 0 0
tracefs /sys/kernel/tracing tracefs rw,nosuid,nodev,noexec,relatime 0 0
sunrpc /run/rpc_pipefs rpc_pipefs rw,relatime 0 0
nfsd /proc/fs/nfsd nfsd rw,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,nosuid,nodev,noexec,relatime 0
0
configfs /sys/kernel/config configfs rw,nosuid,nodev,noexec,relatime 0 0
/dev/sda1 /boot/efi vfat
rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
0 0
192.168.1.200:/u /u nfs4
rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.1.13,local_lock=none,addr=
192.168.1.200 0 0
tmpfs /run/user/1000 tmpfs
rw,nosuid,nodev,relatime,size=803616k,mode=700,uid=1000,gid=1000 0 0
gvfsd-fuse /run/user/1000/gvfs fuse.gvfsd-fuse
rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
/dev/fuse /run/user/1000/doc fuse
rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0

**Comment by alkisg** [comment]
Try to reply on github instead of via email, as the discussion is a lot more readable that way.

I don't see any significant differences on the NFS or the other mount options, except for /home, which is SSHFS.

I would do the following tests:

- NFS mounted home instead of SSHFS: https://ltsp.org/man/ltsp-nfs/
- Check dmesg / journalctl for errors
- Check `free` for RAM issues; enable local swap
